src/options/imports/index_ops.js

```diff
@@ -0,0 +1,96 @@
+import index from 'src/search/search-index'
+import db from 'src/pouchdb'
+const idb = index.db._db
+
+async function downloadChunk(data, counter) {
+    for (const datum of data) {
+        if (datum.value instanceof Map === true) {
+            const serializedValues = []
+            for (const [key, value] of datum.value) {
+                serializedValues.push([key, value])
+            }
+            datum.value = serializedValues
+        }
+        if (datum.key.search('page/') !== -1) {
+            const id = datum.value.id
+            db
+                .get(id)
+                .then(pageData => {
```

Comment:
@arpitgogia RE your [comment](https://github.com/WorldBrain/Memex/pull/224#issuecomment-361494820):
By the time L32 is reached, this stuff (inside the `.then` cb) could still be happening, or could have finished, or not had a chance to start yet - don't really know as it's async (the `db.get` call).  As this is already in an `async function` context, you can explicitly say to `await` the async `db.get` logic (the returned `Promise`) to resolve then continue on with the `browser.downloads.download` stuff only __after__ everything else is done. Eg:

```js
try {
  const pageData = await db.get(id)
  ... // your `datum` mutating code here
} catch (err) {
  ... // handle bad stuff here
  console.error(err)
}
```

Some other notes about this code:
- L7-13 could be simplified to:
```js
if (datum.value instanceof Map || datum.value instanceof Set) {
  datum.value = [...datum.value]
}
```
- L14 `datum.key.search('page/')` should be simplified to `datum.key.startsWith('page/')` - I would also recommend importing the `pageKeyPrefix` constant used throughout the codebase rather than using a literal string here
- rather than iterating over the input `data` and mutating it in place before serializing, create a transformation function that takes in the `data` Array of KVP objects and returns a __new__ array instance of the transformed data (`Array.prototype.map` will do this for you), ready for serializing (or you could just do the serializing inside and have it return a `string`) - this probably the most important IMO, so you're not mutating that `data` input